tests/test_wheelfile.py

```diff
@@ -41,3 +42,16 @@ def test_source_date_epoch(tmp_path, monkeypatch):
     with wheel.wheelfile.WheelFile(path, 'r') as w:
         for entry in w.infolist():
             assert entry.date_time == time.gmtime(epoch)[:6]
+
+
+def test_compression(tmp_path):
+    # create a wheel and test that everything is compressed
+    path = tmp_path / 'test-1.0-py3-any-none.whl'
+    bar = tmp_path / 'bar'
+    bar.write_bytes(b'bar')
+    with mesonpy._wheelfile.WheelFile(path, 'w') as w:
+        w.writestr('foo', b'test')
+        w.write(bar, 'bar')
+    with wheel.wheelfile.WheelFile(path, 'r') as w:
```

Comment:
I would use `ZipFile` directly here. I think that the `RECORD` file in the zip archive can be compressed too, but you patch does not do that, see the `close()` method, however, `WheelFile.inforlist()` does not yield the `RECORD` file and the test misses it.

Reply:
Good point, I've changed it to Zipfile now.

But I don't understand the other parts, close() is patched and inforlist() is never overridden anywhere and shouldn't have made a difference.

Reply:
Yes, sorry, I misread the patch. I assumed the methods patched where `write()` and `writestr()`.